internal/powerplatform/services/tenant_settings/api_tenant_settings.go

```diff
@@ -51,9 +72,72 @@ func (client *TenantSettingsClient) UpdateTenantSettings(ctx context.Context, te
 	values.Add("api-version", "2023-06-01")
 	apiUrl.RawQuery = values.Encode()
 
-	_, err := client.Api.Execute(ctx, "POST", apiUrl.String(), nil, tenantSettings, []int{http.StatusOK}, &tenantSettings)
+	var backendSettings TenantSettingsDto
+	_, err := client.Api.Execute(ctx, "POST", apiUrl.String(), nil, tenantSettings, []int{http.StatusOK}, &backendSettings)
 	if err != nil {
 		return nil, err
 	}
-	return &tenantSettings, nil
+	return &backendSettings, nil
+}
+
+func applyCorrections(planned TenantSettingsDto, actual TenantSettingsDto) *TenantSettingsDto {
+	corrected := filterDto(planned, actual).(*TenantSettingsDto)
+	if planned.PowerPlatform != nil && planned.PowerPlatform.Governance != nil {
+		if planned.PowerPlatform.Governance.EnvironmentRoutingTargetSecurityGroupId != nil && *planned.PowerPlatform.Governance.EnvironmentRoutingTargetSecurityGroupId == ZERO_UUID && corrected.PowerPlatform.Governance.EnvironmentRoutingTargetSecurityGroupId == nil {
+			zu := ZERO_UUID
+			corrected.PowerPlatform.Governance.EnvironmentRoutingTargetSecurityGroupId = &zu
+		}
+
+		if planned.PowerPlatform.Governance.EnvironmentRoutingTargetEnvironmentGroupId != nil && *planned.PowerPlatform.Governance.EnvironmentRoutingTargetEnvironmentGroupId == ZERO_UUID && corrected.PowerPlatform.Governance.EnvironmentRoutingTargetEnvironmentGroupId == nil {
+			zu := ZERO_UUID
+			corrected.PowerPlatform.Governance.EnvironmentRoutingTargetEnvironmentGroupId = &zu
+		}
+	}
+	return corrected
+}
+
+// This function is used to filter out the fields that are not opted in to configuration
+// The backend always returns all properties, but Terraform can only handle the properties that are opted in
+func filterDto(configuredSettings interface{}, backendSettings interface{}) interface{} {
+	configuredType := reflect.TypeOf(configuredSettings)
+	backendType := reflect.TypeOf(backendSettings)
+	if configuredType != backendType {
+		return nil
+	}
+
+	output := reflect.New(configuredType).Interface()
+
+	visibleFields := reflect.VisibleFields(configuredType)
+
+	configuredValue := reflect.ValueOf(configuredSettings)
+	backendValue := reflect.ValueOf(backendSettings)
+
+	for fieldIndex, fieldInfo := range visibleFields {
+		log.Default().Printf("Field: %s", fieldInfo.Name)
+
+		configuredFieldValue := configuredValue.Field(fieldIndex)
+		backendFieldValue := backendValue.Field(fieldIndex)
+		outputField := reflect.ValueOf(output).Elem().Field(fieldIndex)
+
+		if !configuredFieldValue.IsNil() && !backendFieldValue.IsNil() && backendFieldValue.IsValid() && outputField.CanSet() {
+			if configuredFieldValue.Kind() == reflect.Pointer && configuredFieldValue.Elem().Kind() == reflect.Struct {
```

Comment:
we should have at least on unit/acc test, that covers all the "else if" branches of this reflection flow.

Reply:
We will add this as an acceptance test after we merge with new acceptance testing